tests/test_formparsers.py

```diff
@@ -390,10 +394,19 @@ def test_user_safe_decode_ignores_wrong_charset():
     assert result == "abc"
 
 
-def test_missing_boundary_parameter(test_client_factory):
+@pytest.mark.parametrize(
+    "app,expectation",
+    [
+        (app, pytest.raises(MultiPartException)),
+        (Starlette(routes=[Mount("/", app=app)]), does_not_raise()),
+    ],
+)
+def test_missing_boundary_parameter(
+    app, expectation, test_client_factory: typing.Callable[..., TestClient]
+) -> None:
     client = test_client_factory(app)
-    with pytest.raises(KeyError, match="boundary"):
-        client.post(
+    with expectation:
```

Comment:
I'd *personally* probably nudge towards the simpler `pytest.raises(KeyError, match="boundary")` case here, just because it's more obviously readable to me.

The parameterised `expectation` is neat, but also more complex to understand.

Just my personal perspective tho, happy to go either way.

Reply:
The idea was to avoid the creation of two very similar tests. Do you think is clear if I create two tests instead?

Reply:
I think your way or combining an exception and non-exception test into 1 is the cleanest I've seen for that pattern, but if it's
only 2 tests and not 10 I also think making it more explicit would be nice